src/main/java/core/AbstractPlayer.java

```diff
@@ -18,6 +18,7 @@
     public final int getPlayerID() {
         return playerID;
     }
+    public void setPlayerID(int id) {playerID = id;}
```

Comment:
I've backed this out, so that playerID is purely package-level access.

While the only rollout strategy supported is Random, then this is not actually needed (but it is needed for a decent Dominion player). I'll raise a separate issue to deal with the issue around playerID property of an AbstractPlayer.

I personally like the idea of having a Map<Integer, AbstractPlayer> in Game (or perhaps better, AbstractGameState). In terms of opponent modelling, when rolling a state forward, then we:
i) State.copy(ID of perspective player);
ii) State.setOpponentModels(ID of opponent, AbstractPlayer)

and then roll forward. Anyway. Need to discuss this properly.